src/desktop/apps/personalize/client/onboarding.js

```diff
@@ -2,18 +2,32 @@ import React from 'react'
 import ReactDOM from 'react-dom'
 import { Router } from 'react-router'
 import createHistory from 'history/createBrowserHistory'
+import Cookies from 'cookies-js'
 
 import { ContextProvider } from '@artsy/reaction-force/dist/Components/Artsy'
 import { RoutesWithProgressBar } from '../components/RoutesWithProgressBar'
 
 export const init = () => {
   const bootstrapData = window.__BOOTSTRAP__
 
+  let redirectTo = '/'
+  // Check the cookie for a destination
+  if (Cookies.get('destination')) {
+    redirectTo = Cookies.get('destination')
+    Cookies.expire('destination')
+
+    // Also check the redirectTo from query params
+  } else if (bootstrapData.redirectTo) {
+    redirectTo = bootstrapData.redirectTo
```

Comment:
Would it be possible to move this to [this route](https://github.com/artsy/force/blob/master/src/desktop/apps/personalize/routes.js#L5-L20)? This would probably work today because this line would be evaluated before [the client-side redirection gets triggered](https://github.com/artsy/reaction/blob/c1c368d9ca5d8478745a302fc9844f5b5e3debf0/src/Components/Onboarding/Wizard.tsx#L71), which makes this order-dependant. Also, if we switched to server-side redirection for e.g. better analytics support, that would silently break this behaviour.

Reply:
Hmm not sure I understand what the benefit would be. This `bootstrapData.redirectTo` is set from the route via `req.query` [here](https://github.com/artsy/force/blob/master/src/desktop/apps/personalize/routes.js#L41).